Content.Server/Medical/HealthAnalyzerSystem.cs

```diff
@@ -59,15 +110,99 @@ private void OpenUserInterface(EntityUid user, EntityUid analyzer)
             if (!TryComp<ActorComponent>(user, out var actor) || !_uiSystem.TryGetUi(analyzer, HealthAnalyzerUiKey.Key, out var ui))
                 return;
 
-            _uiSystem.OpenUi(ui ,actor.PlayerSession);
+            _uiSystem.OpenUi(ui, actor.PlayerSession);
         }
 
-        public void UpdateScannedUser(EntityUid uid, EntityUid user, EntityUid? target, HealthAnalyzerComponent? healthAnalyzer)
+        /// <summary>
+        /// Mark the entity as having its health analyzed, and link the analyzer to it
+        /// </summary>
+        private void BeginAnalyzingEntity(EntityUid uid, EntityUid healthAnalyzerUid, HealthAnalyzerComponent? component = null, HealthBeingAnalyzedComponent? healthBeingAnalyzedComponent = null)
         {
-            if (!Resolve(uid, ref healthAnalyzer))
+            if (!Resolve(healthAnalyzerUid, ref component))
                 return;
 
-            if (target == null || !_uiSystem.TryGetUi(uid, HealthAnalyzerUiKey.Key, out var ui))
+            if (!Resolve(uid, ref healthBeingAnalyzedComponent, false))
+            {
+                //No other active analyzers, create the component and attach to the scanned entity
+                healthBeingAnalyzedComponent = new HealthBeingAnalyzedComponent();
+                AddComp(uid, healthBeingAnalyzedComponent);
+            }
+
+            if (!healthBeingAnalyzedComponent.ActiveAnalyzers.Contains(healthAnalyzerUid))
+                healthBeingAnalyzedComponent.ActiveAnalyzers.Add(healthAnalyzerUid);
+
+            component.ScannedEntity = uid;
+        }
+
+        /// <summary>
+        /// Remove the analyzer from the active list, and remove the component if it has no active analyzers
+        /// </summary>
+        private void StopAnalyzingEntity(EntityUid uid, EntityUid healthAnalyzerUid, HealthAnalyzerComponent? component = null, HealthBeingAnalyzedComponent? healthBeingAnalyzedComponent = null)
```

Comment:
Could probably replace the params with two `Entity<TComp?>`s

Reply:
Removed the second param.
Im keeping the first as i use it to handle the health analyser being deleted/destroyed.